platforms/pegasus/motion_control.cpp

```diff
@@ -0,0 +1,339 @@
+}
+
+static void pf_encoder_reset(void)
+{
+    qdec_read_and_reset(MOTOR_LEFT);
+    qdec_read_and_reset(MOTOR_RIGHT);
+}
+
+void compute_current_speed_and_pose(cogip::cogip_defs::Polar &current_speed, cogip::cogip_defs::Pose &current_pose)
+{
+    pf_encoder_read(current_speed);
+    odometry_update(current_pose, current_speed, SEGMENT);
+}
+
+
+void pf_motor_drive(const cogip::cogip_defs::Polar &command)
+{
+    int16_t right_command = 0;
+    int16_t left_command = 0;
+
+    if (pf_motion_control_platform_engine.pose_reached() == cogip::motion_control::target_pose_status_t::moving) {
+        // Compute motor commands with Polar motion control result
+        right_command = (int16_t) (command.distance() + command.angle());
+        left_command = (int16_t) (command.distance() - command.angle());
+    }
+    else {
+        // Reset speed PIDs if a pose has been reached (intermediate or final)
+        // Pose PIDs do not need to be reset as they only have Kp (no sum of error)
+        linear_speed_pid.reset();
+        angular_speed_pid.reset();
+
+        // Send message in case of final pose reached only
+        if (pf_motion_control_platform_engine.pose_reached() == cogip::motion_control::target_pose_status_t::reached) {
+            pf_get_uartpb().send_message(cogip::pf::motion_control::pose_reached_uuid);
+        }
+    }
+
+    // Apply motor commands
+    motor_set(MOTOR_DRIVER_DEV(0), MOTOR_LEFT, left_command);
+    motor_set(MOTOR_DRIVER_DEV(0), MOTOR_RIGHT, right_command);
+
+    // Send robot state
+    pf_send_pb_state();
+}
+
+/// Handle pid request command message.
+static void _handle_pid_request([[maybe_unused]] cogip::uartpb::ReadBuffer & buffer)
+{
+    // Send PIDs
+    pf_send_pids();
+}
+
+void pf_init_motion_control(void)
+{
+    // Init motor driver
+    motor_driver_init(MOTOR_DRIVER_DEV(0));
+
+    // Setup qdec periphereal
+    int error = qdec_init(QDEC_DEV(MOTOR_LEFT), QDEC_MODE, NULL, NULL);
+    if (error) {
+        printf("QDEC %u not initialized, error=%d !!!\n", MOTOR_LEFT, error);
+    }
+    error = qdec_init(QDEC_DEV(MOTOR_RIGHT), QDEC_MODE, NULL, NULL);
+    if (error) {
+        printf("QDEC %u not initialized, error=%d !!!\n", MOTOR_RIGHT, error);
+    }
+
+    // Init odometry
+    odometry_setup(cogip::pf::motion_control::wheels_distance / cogip::pf::motion_control::pulse_per_mm);
+
+    //TODO: update
+    //ctrl_set_anti_blocking_on(pf_get_ctrl(), TRUE);
+
+    // Init controllers
+    pf_quadpid_meta_controller = pf_quadpid_meta_controller_init();
+
+    // Associate a controller to the engine
+    pf_motion_control_platform_engine.set_controller(pf_quadpid_meta_controller);
+
+    // Register pid request command
+    pf_get_uartpb().register_message_handler(
+        cogip::pf::motion_control::pid_request_uuid,
+        cogip::uartpb::message_handler_t::create<_handle_pid_request>()
+    );
+}
+
+} // namespace actuators
+
+} // namespace pf
+
+} // namespace cogip
```

Comment:
Missing empty line at EOF.

Reply:
weird, got them but in a further commit maybe